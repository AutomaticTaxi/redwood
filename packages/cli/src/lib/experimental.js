import path from 'path'

import { getConfig, getPaths, resolveFile } from '@redwoodjs/project-config'

/**
 * Checks to see if the user has the experimental server file present at api/src/server.ts|js and enabled in redwood.toml
 *
 * @returns {boolean} true if the experimental server file is enabled and present
 */
export function isUsingExperimentalServerFile() {
  return (
    getConfig().experimental?.serverFile?.enabled &&
    resolveFile(path.join(getPaths().api.src, 'server'))
  )
}
