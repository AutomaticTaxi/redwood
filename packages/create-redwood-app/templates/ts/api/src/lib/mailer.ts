import { Mailer } from '@redwoodjs/mailer-core'
import { InMemoryMailHandler } from '@redwoodjs/mailer-handler-in-memory'
import { NodemailerMailHandler } from '@redwoodjs/mailer-handler-nodemailer'
import { StudioMailHandler } from '@redwoodjs/mailer-handler-studio'
import { ReactEmailRenderer } from '@redwoodjs/mailer-renderer-react-email'

import { logger } from './logger'

export const mailer = new Mailer({
  // Handlers to send our emails out to the world
  handling: {
    handlers: {
      inMemory: new InMemoryMailHandler(),
      studio: new StudioMailHandler(),
      // TODO: Update this handler config or switch it out for a different handler completely
      nodemailer: new NodemailerMailHandler({
        transport: {
          host: 'localhost',
          port: 4319,
          secure: false,
        },
      }),
    },
    // This is our default handler when in production only
    default: 'nodemailer',
  },

  // Renderers to convert our templates into HTML/text
  rendering: {
    renderers: {
      reactEmail: new ReactEmailRenderer(),
    },
    default: 'reactEmail',
  },

  // Specific handlers for testing and development time
  test: {
    handler: 'inMemory',
    // when: process.env.NODE_ENV === 'test',
  },
  development: {
    handler: 'studio',
    // when: process.env.NODE_ENV !== 'production',
  },

  // Default send options
  defaults: {
    // replyTo: 'noreply@example.com',
  },

  logger,
})
