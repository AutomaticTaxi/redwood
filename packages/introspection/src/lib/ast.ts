import fs from 'node:fs'
import path from 'node:path'

import { parse as babelParse } from '@babel/parser'
import type { ParserPlugin } from '@babel/parser'
import traverse from '@babel/traverse'
import type {
  JSXElement,
  ExportNamedDeclaration,
  File,
  VariableDeclarator,
  ExportDefaultDeclaration,
} from '@babel/types'
import {
  assertJSXOpeningElement,
  assertJSXIdentifier,
  assertIdentifier,
  isJSXAttribute,
  isJSXSpreadAttribute,
  isVariableDeclaration,
  isVariableDeclarator,
  isIdentifier,
} from '@babel/types'

import { getPaths } from '@redwoodjs/project-config'

export function getASTFromFile(filepath: string) {
  const code = fs.readFileSync(filepath, { encoding: 'utf8', flag: 'r' })

  // use jsx plugin for web files, because in JS, the .jsx extension is not used
  const isJsxFile =
    path.extname(filepath).match(/[jt]sx$/) ||
    path.parse(filepath).dir.startsWith(getPaths().web.base)

  const plugins = [
    'typescript',
    'nullishCoalescingOperator',
    'objectRestSpread',
    isJsxFile && 'jsx',
  ].filter(Boolean) as ParserPlugin[]

  return babelParse(code, {
    sourceType: 'module',
    plugins,
  })
}

export function getSpecificNamedExportDeclarations(ast: File, names: string[]) {
  const desiredNamedExports: Map<string, VariableDeclarator> = new Map()
  const allNamedExports: ExportNamedDeclaration[] = []
  traverse(ast, {
    ExportNamedDeclaration: (path) => {
      allNamedExports.push(path.node)
    },
  })
  allNamedExports.forEach((namedExport) => {
    if (
      namedExport.declaration != null &&
      isVariableDeclaration(namedExport.declaration) &&
      isVariableDeclarator(namedExport.declaration.declarations[0]) &&
      isIdentifier(namedExport.declaration.declarations[0].id) &&
      names.includes(namedExport.declaration.declarations[0].id.name)
    ) {
      desiredNamedExports.set(
        namedExport.declaration.declarations[0].id.name,
        namedExport.declaration.declarations[0]
      )
    }
  })
  return desiredNamedExports
}

export function getExportDefaultDeclaration(ast: File) {
  let defaultExport: ExportDefaultDeclaration | undefined
  traverse(ast, {
    ExportDefaultDeclaration: (path) => {
      defaultExport = path.node
    },
  })
  return defaultExport
}

export function getVariableDeclarator(ast: File, name: string) {
  let variableDeclarator: VariableDeclarator | undefined
  traverse(ast, {
    VariableDeclarator: (path) => {
      if (isIdentifier(path.node.id) && path.node.id.name === name) {
        variableDeclarator = path.node
      }
    },
  })
  return variableDeclarator
}

export function getJSXElementAttributes(element: JSXElement) {
  const attributes: Map<string, string> = new Map()
  assertJSXOpeningElement(element.openingElement)
  if (element.openingElement.attributes.length > 0) {
    element.openingElement.attributes.forEach((attribute) => {
      if (isJSXAttribute(attribute)) {
        assertJSXIdentifier(attribute.name)
        const key = attribute.name.name
        let value: string
        switch (attribute.value?.type) {
          case 'StringLiteral':
            value = attribute.value.value
            break
          case 'JSXExpressionContainer':
            assertIdentifier(attribute.value.expression)
            value = attribute.value.expression.name
            break
          case undefined: // Accept this because it could be a attribute with no value, i.e. notfound
            value = 'true'
            break
          default:
            throw new Error(
              `Unknown value type found while trying to process the ${key} property when extracting all attributes`
            )
        }
        attributes.set(key, value)
      } else if (isJSXSpreadAttribute(attribute)) {
        throw new Error(
          'JSXSpreadAttribute type is not yet supported when extracting all attributes'
        )
      } else {
        throw new Error(
          'Uknown attribute type found while trying to extract all attributes'
        )
      }
    })
  }
  return attributes
}
