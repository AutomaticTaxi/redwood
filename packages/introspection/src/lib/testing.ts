import { ensurePosixPath } from '@redwoodjs/project-config'

export function stripAndFormatPathForTesting(
  filepath: string,
  projectFilepath: string
) {
  if (filepath.startsWith(projectFilepath)) {
    filepath = ensurePosixPath(filepath.substring(projectFilepath.length))
  }
  return filepath
}
