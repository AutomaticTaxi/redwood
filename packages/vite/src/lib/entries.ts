import path from 'node:path'

import type { PagesDependency } from '@redwoodjs/project-config'
import {
  ensurePosixPath,
  getPaths,
  processPagesDir,
} from '@redwoodjs/project-config'

const getPathRelativeToSrc = (maybeAbsolutePath: string) => {
  // If the path is already relative
  if (!path.isAbsolute(maybeAbsolutePath)) {
    return maybeAbsolutePath
  }

  return `./${path.relative(getPaths().web.src, maybeAbsolutePath)}`
}

const withRelativeImports = (page: PagesDependency) => {
  return {
    ...page,
    relativeImport: ensurePosixPath(getPathRelativeToSrc(page.importPath)),
  }
}

export function getEntries() {
  return processPagesDir()
    .map(withRelativeImports)
    .reduce(
      (acc, page) => {
        acc[page.importName] = page.path
        return acc
      },
      {} as Record<string, string>,
    )
}

export async function getEntriesFromDist(): Promise<Record<string, string>> {
  const entriesDist = getPaths().web.distRscEntries
  const { serverEntries } = await import(`file://${entriesDist}`)
  return serverEntries
}
