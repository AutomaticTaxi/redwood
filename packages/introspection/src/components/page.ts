import fs from 'fs'
import path from 'path'

import traverse from '@babel/traverse'

import { getPaths } from '@redwoodjs/project-config'

import { getASTFromFile } from '../lib/ast'

import type {
  RedwoodIntrospectionError,
  RedwoodIntrospectionWarning,
} from './introspection'
import { RedwoodIntrospectionComponent } from './introspection'

export class RedwoodPage extends RedwoodIntrospectionComponent {
  readonly type = 'page'

  readonly hasDefaultExport: boolean

  private constructor(filepath: string) {
    super(filepath)

    const ast = getASTFromFile(this.filepath)

    // Check if there is a default export
    let hasDefaultExport = false
    traverse(ast, {
      ExportDefaultDeclaration: () => {
        hasDefaultExport = true
      },
    })
    this.hasDefaultExport = hasDefaultExport
  }

  getErrors(): RedwoodIntrospectionError[] {
    const errors: RedwoodIntrospectionError[] = []

    // Ensure that there is a default export
    if (!this.hasDefaultExport) {
      errors.push({
        component: {
          type: this.type,
          filepath: this.filepath,
          name: this.name,
        },
        message: 'No default export found',
      })
    }

    return errors
  }

  getWarnings(): RedwoodIntrospectionWarning[] {
    // No warnings for pages
    return []
  }

  static parsePage(filepath: string) {
    return new RedwoodPage(filepath)
  }

  static parsePages(directory: string = getPaths().web.pages) {
    const pages: RedwoodPage[] = []

    // TODO: Confirm this the condition to detect a page
    // Pages must be defined within a file which ends with `Page.{js, jsx, tsx}`
    const getPageFiles = (directory: string) => {
      const pageFiles: string[] = []
      const directoryContents = fs.readdirSync(directory)
      directoryContents.forEach((content) => {
        const stat = fs.lstatSync(path.join(directory, content))
        if (stat.isDirectory()) {
          pageFiles.push(...getPageFiles(path.join(directory, content)))
        } else if (stat.isFile()) {
          if (content.match(/.+Page\.(js|jsx|tsx)$/)) {
            pageFiles.push(path.join(directory, content))
          }
        }
      })
      return pageFiles
    }

    const pageFiles = getPageFiles(directory)
    pageFiles.forEach((pagePath) => {
      pages.push(this.parsePage(pagePath))
    })

    return pages
  }
}
