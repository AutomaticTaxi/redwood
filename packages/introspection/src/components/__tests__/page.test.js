import path from 'node:path'

import { stripAndFormatPathForTesting } from '../../lib/testing'
import { RedwoodPage } from '../page'

const FIXTURE_PATH = path.join(
  __dirname,
  '..',
  '..',
  '..',
  '..',
  '..',
  '__fixtures__'
)
const FIXTURE_PROJECTS = [
  'empty-project',
  'example-todo-main',
  'example-todo-main-with-errors',
  'test-project',
]

describe.each(FIXTURE_PROJECTS)(
  "From within the '%s' fixture",
  (PROJECT_NAME) => {
    // Setup RWJS_CWD to point to the fixture project
    const PROJECT_PATH = path.join(FIXTURE_PATH, PROJECT_NAME)
    const RWJS_CWD = process.env.RWJS_CWD
    beforeAll(() => {
      process.env.RWJS_CWD = PROJECT_PATH
    })
    afterAll(() => {
      process.env.RWJS_CWD = RWJS_CWD
    })

    it('returns the correct pages', () => {
      const pages = RedwoodPage.parsePages()
      expect(pages.length).toMatchSnapshot('pages count')
      for (let i = 0; i < pages.length; i++) {
        const page = pages[i]
        page.filepath = stripAndFormatPathForTesting(
          page.filepath,
          PROJECT_PATH
        )
        expect(page).toMatchSnapshot(`${page.filepath} content`)
        expect(page.getErrors()).toMatchSnapshot(`${page.filepath} errors`)
        expect(page.getWarnings()).toMatchSnapshot(`${page.filepath} warnings`)
      }
    })
  }
)
