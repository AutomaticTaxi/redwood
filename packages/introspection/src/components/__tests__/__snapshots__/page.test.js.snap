// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`From within the 'empty-project' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/FatalErrorPage/FatalErrorPage.tsx",
  "hasDefaultExport": true,
  "name": "FatalErrorPage",
  "type": "page",
}
`;

exports[`From within the 'empty-project' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.tsx errors 1`] = `[]`;

exports[`From within the 'empty-project' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.tsx warnings 1`] = `[]`;

exports[`From within the 'empty-project' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/NotFoundPage/NotFoundPage.tsx",
  "hasDefaultExport": true,
  "name": "NotFoundPage",
  "type": "page",
}
`;

exports[`From within the 'empty-project' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.tsx errors 1`] = `[]`;

exports[`From within the 'empty-project' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.tsx warnings 1`] = `[]`;

exports[`From within the 'empty-project' fixture returns the correct pages: pages count 1`] = `2`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/BarPage/BarPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/BarPage/BarPage.tsx",
  "hasDefaultExport": true,
  "name": "BarPage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/BarPage/BarPage.tsx errors 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/BarPage/BarPage.tsx warnings 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.js content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/FatalErrorPage/FatalErrorPage.js",
  "hasDefaultExport": true,
  "name": "FatalErrorPage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.js errors 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.js warnings 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/FooPage/FooPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/FooPage/FooPage.tsx",
  "hasDefaultExport": true,
  "name": "FooPage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/FooPage/FooPage.tsx errors 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/FooPage/FooPage.tsx warnings 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/HomePage/HomePage.tsx",
  "hasDefaultExport": true,
  "name": "HomePage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.tsx errors 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.tsx warnings 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.js content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/NotFoundPage/NotFoundPage.js",
  "hasDefaultExport": true,
  "name": "NotFoundPage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.js errors 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.js warnings 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/PrivatePage/PrivatePage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/PrivatePage/PrivatePage.tsx",
  "hasDefaultExport": true,
  "name": "PrivatePage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/PrivatePage/PrivatePage.tsx errors 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/PrivatePage/PrivatePage.tsx warnings 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/TypeScriptPage/TypeScriptPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/TypeScriptPage/TypeScriptPage.tsx",
  "hasDefaultExport": true,
  "name": "TypeScriptPage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/TypeScriptPage/TypeScriptPage.tsx errors 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/TypeScriptPage/TypeScriptPage.tsx warnings 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/admin/EditUserPage/EditUserPage.jsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/admin/EditUserPage/EditUserPage.jsx",
  "hasDefaultExport": true,
  "name": "EditUserPage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/admin/EditUserPage/EditUserPage.jsx errors 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: /web/src/pages/admin/EditUserPage/EditUserPage.jsx warnings 1`] = `[]`;

exports[`From within the 'example-todo-main' fixture returns the correct pages: pages count 1`] = `8`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.js content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/FatalErrorPage/FatalErrorPage.js",
  "hasDefaultExport": true,
  "name": "FatalErrorPage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.js errors 1`] = `[]`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.js warnings 1`] = `[]`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.js content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/HomePage/HomePage.js",
  "hasDefaultExport": true,
  "name": "HomePage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.js errors 1`] = `[]`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.js warnings 1`] = `[]`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.js content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/NotFoundPage/NotFoundPage.js",
  "hasDefaultExport": true,
  "name": "NotFoundPage",
  "type": "page",
}
`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.js errors 1`] = `[]`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.js warnings 1`] = `[]`;

exports[`From within the 'example-todo-main-with-errors' fixture returns the correct pages: pages count 1`] = `3`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/AboutPage/AboutPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/AboutPage/AboutPage.tsx",
  "hasDefaultExport": true,
  "name": "AboutPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/AboutPage/AboutPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/AboutPage/AboutPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/BlogPostPage/BlogPostPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/BlogPostPage/BlogPostPage.tsx",
  "hasDefaultExport": true,
  "name": "BlogPostPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/BlogPostPage/BlogPostPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/BlogPostPage/BlogPostPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/ContactPage/ContactPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/Contact/ContactPage/ContactPage.tsx",
  "hasDefaultExport": true,
  "name": "ContactPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/ContactPage/ContactPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/ContactPage/ContactPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/ContactsPage/ContactsPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/Contact/ContactsPage/ContactsPage.tsx",
  "hasDefaultExport": true,
  "name": "ContactsPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/ContactsPage/ContactsPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/ContactsPage/ContactsPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/EditContactPage/EditContactPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/Contact/EditContactPage/EditContactPage.tsx",
  "hasDefaultExport": true,
  "name": "EditContactPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/EditContactPage/EditContactPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/EditContactPage/EditContactPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/NewContactPage/NewContactPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/Contact/NewContactPage/NewContactPage.tsx",
  "hasDefaultExport": true,
  "name": "NewContactPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/NewContactPage/NewContactPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Contact/NewContactPage/NewContactPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ContactUsPage/ContactUsPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/ContactUsPage/ContactUsPage.tsx",
  "hasDefaultExport": true,
  "name": "ContactUsPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ContactUsPage/ContactUsPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ContactUsPage/ContactUsPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/DoublePage/DoublePage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/DoublePage/DoublePage.tsx",
  "hasDefaultExport": true,
  "name": "DoublePage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/DoublePage/DoublePage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/DoublePage/DoublePage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/FatalErrorPage/FatalErrorPage.tsx",
  "hasDefaultExport": true,
  "name": "FatalErrorPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/FatalErrorPage/FatalErrorPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ForgotPasswordPage/ForgotPasswordPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/ForgotPasswordPage/ForgotPasswordPage.tsx",
  "hasDefaultExport": true,
  "name": "ForgotPasswordPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ForgotPasswordPage/ForgotPasswordPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ForgotPasswordPage/ForgotPasswordPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/HomePage/HomePage.tsx",
  "hasDefaultExport": true,
  "name": "HomePage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/HomePage/HomePage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/LoginPage/LoginPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/LoginPage/LoginPage.tsx",
  "hasDefaultExport": true,
  "name": "LoginPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/LoginPage/LoginPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/LoginPage/LoginPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/NotFoundPage/NotFoundPage.tsx",
  "hasDefaultExport": true,
  "name": "NotFoundPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/NotFoundPage/NotFoundPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/EditPostPage/EditPostPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/Post/EditPostPage/EditPostPage.tsx",
  "hasDefaultExport": true,
  "name": "EditPostPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/EditPostPage/EditPostPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/EditPostPage/EditPostPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/NewPostPage/NewPostPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/Post/NewPostPage/NewPostPage.tsx",
  "hasDefaultExport": true,
  "name": "NewPostPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/NewPostPage/NewPostPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/NewPostPage/NewPostPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/PostPage/PostPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/Post/PostPage/PostPage.tsx",
  "hasDefaultExport": true,
  "name": "PostPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/PostPage/PostPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/PostPage/PostPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/PostsPage/PostsPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/Post/PostsPage/PostsPage.tsx",
  "hasDefaultExport": true,
  "name": "PostsPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/PostsPage/PostsPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/Post/PostsPage/PostsPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ProfilePage/ProfilePage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/ProfilePage/ProfilePage.tsx",
  "hasDefaultExport": true,
  "name": "ProfilePage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ProfilePage/ProfilePage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ProfilePage/ProfilePage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ResetPasswordPage/ResetPasswordPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/ResetPasswordPage/ResetPasswordPage.tsx",
  "hasDefaultExport": true,
  "name": "ResetPasswordPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ResetPasswordPage/ResetPasswordPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/ResetPasswordPage/ResetPasswordPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/SignupPage/SignupPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/SignupPage/SignupPage.tsx",
  "hasDefaultExport": true,
  "name": "SignupPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/SignupPage/SignupPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/SignupPage/SignupPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/WaterfallPage/WaterfallPage.tsx content 1`] = `
RedwoodPage {
  "filepath": "/web/src/pages/WaterfallPage/WaterfallPage.tsx",
  "hasDefaultExport": true,
  "name": "WaterfallPage",
  "type": "page",
}
`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/WaterfallPage/WaterfallPage.tsx errors 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: /web/src/pages/WaterfallPage/WaterfallPage.tsx warnings 1`] = `[]`;

exports[`From within the 'test-project' fixture returns the correct pages: pages count 1`] = `21`;
